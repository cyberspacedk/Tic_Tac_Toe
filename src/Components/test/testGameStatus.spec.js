/* eslint-disable */
import React from "react";
import { Provider } from "react-redux";
import { shallow, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import configureStore from "redux-mock-store";
import GameStatusConnected, { GameStatus } from "../GameStatus";

describe("GameStatus. show game results", () => {

  // 1- check. render
  // 2- snapshot
  // 3- check. props from store. Count played games
  // 4- check. props from store. Player-1 score
  // 5- check. props from store. Player-2 score
  // 6- check. is component render all chilren

  const mockStore = configureStore();
  let store, myComponent;

  const initialState = {
    playedGame: 7,
    playerScore: {
      player1: 2,
      player2: 5
    }
  };

  beforeEach(() => {
    store = mockStore(initialState);
    myComponent = mount(
      <Provider store={store}>
        <GameStatusConnected />
      </Provider>
    );
  });

  it("Check. render component ", () => {
    expect(myComponent.length).toEqual(1);
  });

  it("GameStatus snapshot", () => {
    expect(shallowToJson(myComponent)).toMatchSnapshot();
  });

  it("Check. props from store. Count played games", () => {
    expect(myComponent.find('[name="sets-played"]').text()).toEqual(
      "Sets played: 7"
    );
  });

  it("Check. props from store. Player-1 score", () => {
    expect(myComponent.find('[name="player-1"]').text()).toEqual(
      "Player 1 wins : 2"
    );
  });

  it("Check. props from store. Player-2 score", () => {
    expect(myComponent.find('[name="player-2"]').text()).toEqual(
      "Player 2 wins : 5"
    );
  });

  it("Check. is component render all chilren", () => {
    expect(myComponent.find("p").length).toEqual(3);
  });
 
});
