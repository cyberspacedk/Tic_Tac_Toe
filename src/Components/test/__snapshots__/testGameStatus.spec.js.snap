// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameStatus. show game results GameStatus snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ConnectFunction>
    <GameStatus
      playedGame={7}
      playerOne={[Function]}
      playerScore={
        Object {
          "player1": 2,
          "player2": 5,
        }
      }
      playerTwo={[Function]}
      storePassedGame={[Function]}
    >
      <styled.div>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "dTilpC",
                "rules": Array [
                  "
	background-color:#3c3c3c;
	padding:10px 0 20px;
	width:100vw;
	text-align:center;
	color: #fff;
	font-size: 1.2rem;
	box-shadow: 1px -4px 13px 0px #676767; 
	p:nth-child(1){
    font-size: 2rem;
	}
	p:nth-child(2){
    color: green
	}
	p:nth-child(3){
    color: orange
	}
",
                ],
              },
              "displayName": "styled.div",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <div
            className="sc-bdVaJa dTilpC"
          >
            <p
              name="sets-played"
            >
              Sets played: 
              7
            </p>
            <p
              name="player-1"
            >
              Player 1 wins : 
              2
            </p>
            <p
              name="player-2"
            >
              Player 2 wins : 
              5
            </p>
          </div>
        </StyledComponent>
      </styled.div>
    </GameStatus>
  </ConnectFunction>
</Provider>
`;
